src/main/kotlin/xyz/crossward/entities/Entry.kt

```diff
@@ -27,7 +27,10 @@ data class EntryId(
 private val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd")
 
 fun Entry.localDate(): LocalDate? {
-    return LocalDate.parse(this.date, formatter)
+    if (this.date != null) {
```

Comment:
I think the most kotlin way to do this would be:
```Kotlin
fun Entry.localDate(): LocalDate? =
    this.let { LocalDate.parse(this.date, formatter) } ?: null
```

Reply:
This was close but still produced a NullPointerException, take a look at what I came up with.

Reply:
Looks good